src/algorithm/native/to_wkt.rs

```diff
@@ -0,0 +1,282 @@
+use crate::geo_traits::*;
+
+/// Converts the input to WKT representation.
+pub trait ToWKT {
+    fn to_wkt(&self) -> String;
+}
+
+impl<G: GeometryTrait<T = f64>> ToWKT for G {
+    fn to_wkt(&self) -> String {
+        match self.as_type() {
+            GeometryType::Point(point) => {
+                let mut wkt = String::from("POINT");
+
+                let x = point.x();
+                let y = point.y();
+
+                // handle NaN, may should hapen when reading
+                if x.is_nan() && y.is_nan() {
+                    wkt.push_str(" EMPTY");
+                    return wkt;
+                }
+
+                if point.dim() == 3 {
+                    wkt.push('Z')
+                }
```

Comment:
As of today this split between the `Dimension` enum and the usize `dim()` in the trait is intentional.

In geoarrow-rs we have a split between the physical layout and the semantic meaning of those dimensions. Arrays are parameterized by a const generic `D: usize`. But in the future to support `XYM` we'll also need metadata about what the physical dimensions represent.

This is separate from geo-traits. The prototype in this repo of multi-dimensional geo-traits is **physical only**. So it's intentional for the dimension to return a `usize`. Ref https://github.com/georust/geo/pull/1157#issuecomment-2014843423, https://github.com/georust/geo/pull/1157#issuecomment-2033494580, https://github.com/georust/geo/pull/1157#issuecomment-2033501634.


The split between physical layout and semantic meaning also made it considerably easier to implement, because `usize` is a const generic, whereas I struggled to get things working with Dimension as a generic. (I don't remember the exact reason off hand).